src/store/GlobalState.js

```diff
@@ -0,0 +1,155 @@
+import { createContext, useEffect, useReducer } from 'react';
+import { collection, doc, addDoc, deleteDoc, updateDoc, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+import appReducer from './appReducer';
+import { getCurrentMonth, getUKFormattedDate } from '../utils/helpers';
+
+const currentMonth = getCurrentMonth();
+
+const initialState = {
+  expenses: [],
+  selectedMonthExpenses: [],
+  currentMonth: currentMonth,
+  selectedMonth: currentMonth,
+  lastAddedExpense: {},
+  expenseToDelete: {},
+  expenseToEdit: {},
```

Comment:
these two do not seem like global at all. my guess is that they are highly local.